src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Helper/Base64ValidatorHelper.cs

```diff
@@ -9,23 +9,29 @@ internal static partial class Base64Helper
     {
         internal static bool IsValid<T, TBase64Validatable>(TBase64Validatable validatable, ReadOnlySpan<T> base64Text, out int decodedLength)
             where TBase64Validatable : IBase64Validatable<T>
+            where T : struct
         {
             int length = 0, paddingCount = 0;
+            T lastChar = default;
 
             if (!base64Text.IsEmpty)
             {
 #if NET
-                while (true)
+                while (!base64Text.IsEmpty)
                 {
-
                     int index = validatable.IndexOfAnyExcept(base64Text);
                     if ((uint)index >= (uint)base64Text.Length)
                     {
                         length += base64Text.Length;
+                        lastChar = base64Text[base64Text.Length - 1];
                         break;
                     }
 
                     length += index;
+                    if (index > 0)
+                    {
+                        lastChar = base64Text[index - 1];
+                    }
```

Comment:
Are there any perf consequences to these changes?

Reply:
Both APIs did not show noticeable impact with the perf tests I run, but the input doesn't include whitespaces, I will try with an input with whitespaces

Reply:
Around 0.03 percent degradation with whitespace

| Method                       | Toolchain           | Mean       | Error     | StdDev     | Median     | Ratio | RatioSD |
|----------------------------- |-------------------- |-----------:|----------:|-----------:|-----------:|------:|--------:|
| Base64IsValid                | \lastPR\corerun.exe |   5.734 us | 0.0576 us |  0.0511 us |   5.729 us |  1.01 |    0.01 |
| Base64IsValid                | \main\corerun.exe   |   5.665 us | 0.0159 us |  0.0133 us |   5.665 us |  1.00 |    0.00 |
|                              |                     |            |           |            |            |       |         |
| Base64IsValidWithWhiteSpaces | \lastPR\corerun.exe | 173.792 us | 0.5332 us |  0.4988 us | 173.837 us |  1.03 |    0.00 |
| Base64IsValidWithWhiteSpaces | \main\corerun.exe   | 169.125 us | 0.4375 us |  0.3879 us | 169.179 us |  1.00 |    0.00 |
|                              |                     |            |           |            |            |       |         |
| Base64Decode                 | \lastPR\corerun.exe |   6.036 us | 0.0236 us |  0.0221 us |   6.032 us |  1.00 |    0.01 |
| Base64Decode                 | \main\corerun.exe   |   6.044 us | 0.0276 us |  0.0231 us |   6.031 us |  1.00 |    0.00 |
|                              |                     |            |           |            |            |       |         |
| Base64DecodeInPlace          | \lastPR\corerun.exe | 104.810 us | 2.0912 us |  5.4724 us | 103.150 us |  0.94 |    0.10 |
| Base64DecodeInPlace          | \main\corerun.exe   | 112.553 us | 3.9723 us | 11.3972 us | 109.000 us |  1.00 |    0.00 |